# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/patches/pkg_iproute2.patch
# Copyright (C) 2004 - 2007 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- iproute2-2.6.9.orig/Makefile	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/Makefile	2004-12-23 18:45:12.000000000 -0300
@@ -10,7 +10,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=-lcompat
 ADDLIB=
 
 #options for decnet
@@ -27,7 +27,7 @@
 
 LDLIBS += -L../lib -lnetlink ../lib/libutil.a
 
-SUBDIRS=lib ip tc misc netem genl
+SUBDIRS=lib ip
 
 LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
 
--- iproute2-2.6.9.orig/ip/ipaddress.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/ip/ipaddress.c	2004-12-23 18:50:12.000000000 -0300
@@ -25,8 +25,9 @@
 #include <string.h>
 #include <fnmatch.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
@@ -34,6 +35,39 @@
 #include "ll_map.h"
 #include "ip_common.h"
 
+struct net_device_stats
+{
+        unsigned long   rx_packets;             /* total packets received       */
+        unsigned long   tx_packets;             /* total packets transmitted    */
+        unsigned long   rx_bytes;               /* total bytes received         */
+        unsigned long   tx_bytes;               /* total bytes transmitted      */
+        unsigned long   rx_errors;              /* bad packets received         */
+        unsigned long   tx_errors;              /* packet transmit problems     */
+        unsigned long   rx_dropped;             /* no space in linux buffers    */
+        unsigned long   tx_dropped;             /* no space available in linux  */
+        unsigned long   multicast;              /* multicast packets received   */
+        unsigned long   collisions;
+
+        /* detailed rx_errors: */
+        unsigned long   rx_length_errors;
+        unsigned long   rx_over_errors;         /* receiver ring buff overflow  */
+        unsigned long   rx_crc_errors;          /* recved pkt with crc error    */
+        unsigned long   rx_frame_errors;        /* recv'd frame alignment error */
+        unsigned long   rx_fifo_errors;         /* recv'r fifo overrun          */
+        unsigned long   rx_missed_errors;       /* receiver missed packet       */
+
+        /* detailed tx_errors */
+        unsigned long   tx_aborted_errors;
+        unsigned long   tx_carrier_errors;
+        unsigned long   tx_fifo_errors;
+        unsigned long   tx_heartbeat_errors;
+        unsigned long   tx_window_errors;
+
+        /* for cslip etc */
+        unsigned long   rx_compressed;
+        unsigned long   tx_compressed;
+};
+
 static struct
 {
 	int ifindex;
--- iproute2-2.6.9.orig/ip/iplink.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/ip/iplink.c	2004-12-23 18:44:07.000000000 -0300
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- iproute2-2.6.9.orig/ip/ipmaddr.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/ip/ipmaddr.c	2004-12-23 18:44:09.000000000 -0300
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- iproute2-2.6.9.orig/ip/ipmroute.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/ip/ipmroute.c	2004-12-23 18:44:10.000000000 -0300
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
--- iproute2-2.6.9.orig/ip/iptunnel.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/ip/iptunnel.c	2004-12-23 18:44:06.000000000 -0300
@@ -16,6 +16,9 @@
  * Phil Karn <karn@ka9q.ampr.org>	990408:	"pmtudisc" flag
  */
 
+#define _LINUX_TYPES_H
+#include <asm/byteorder.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- iproute2-2.6.9.orig/lib/ll_addr.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/lib/ll_addr.c	2004-12-23 18:52:29.000000000 -0300
@@ -21,8 +21,8 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
--- iproute2-2.6.9.orig/lib/ll_proto.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/lib/ll_proto.c	2004-12-23 18:53:32.000000000 -0300
@@ -21,8 +21,9 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
+#include <net/if_ether.h>
 #include <linux/sockios.h>
 
 #include "utils.h"
--- iproute2-2.6.9.orig/lib/ll_types.c	2004-12-23 18:40:20.000000000 -0300
+++ iproute2-2.6.9/lib/ll_types.c	2004-12-23 18:54:26.000000000 -0300
@@ -21,8 +21,8 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
+//#include <linux/netdevice.h>
+#include <net/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
